monai/networks/utils.py

```diff
@@ -748,6 +748,70 @@ def convert_to_torchscript(
     return script_module
 
 
+def _onnx_trt_compile(
+    onnx_model,
+    min_shape: Sequence[int],
+    opt_shape: Sequence[int],
+    max_shape: Sequence[int],
+    device: int,
+    precision: str,
+    input_names: Sequence[str] | None,
+    output_names: Sequence[str] | None,
+):
+    """
+    onnx_model: the source ONNX model to compile.
+    min_shape: the minimum input shape of the converted TensorRT model.
+    opt_shape: the optimization input shape of the model, on which the TensorRT optimizes.
+    max_shape: the maximum input shape of the converted TensorRT model.
+    device: the target GPU index to convert and verify the model.
+    precision: the weight precision of the converted TensorRT engine-based TorchScript model.
+        Should be 'fp32' or 'fp16'.
+    input_names: optional input names of the ONNX model.
+    output_names: optional output names of the ONNX model.
+    """
+    trt, _ = optional_import("tensorrt", "8.5.3")
+    torch_tensorrt, _ = optional_import("torch_tensorrt", "1.4.0")
+
+    input_shapes = (min_shape, opt_shape, max_shape)
+    # default to an empty list to fit the `torch_tensorrt.ts.embed_engine_in_new_module` function.
+    input_names = [] if not input_names else input_names
+    output_names = [] if not output_names else output_names
+
+    # set up the TensorRT builder
+    torch_tensorrt.set_device(device)
+    logger = trt.Logger(trt.Logger.WARNING)
+    builder = trt.Builder(logger)
+    network = builder.create_network(1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH))
+    profile = builder.create_optimization_profile()
+    if input_names:
+        profile.set_shape(input_names[0], *input_shapes)
+
+    # parse the ONNX model
+    parser = trt.OnnxParser(network, logger)
+    success = parser.parse(onnx_model.SerializeToString())
+    for idx in range(parser.num_errors):
+        print(parser.get_error(idx))
```

Comment:
Can we raise an exception here directly?

Thanks.

Reply:
Ok, I can delete the for loop here and update the exception content here. It will omit some details of the error, but it should be enough for this function.

Thanks,
Bin